### PR TITLE
Travis to wercker

### PR DESCRIPTION
**Pull request by tkrs**

### CODE REVIEW

File: src/main/scala/src/org/aerospiker/Operation.scala [hunk 1]
```diff
@@ -43,12 +43,15 @@ trait Operation { self: BaseClient =>
       ).trans
     })
 
-  private def futuring[B, A](e: AerospikeException => B)(f: () => A): EitherT[Future, B, A] =
+  private def futuring[B, A](e: AerospikeException => B)(f: () => Option[A]): EitherT[Future, B, A] =
```
Comment (yanana) [inline src/main/scala/src/org/aerospiker/Operation.scala]: By fixing a little verbose pattern matching and literally strange error message:

``` scala
  // this code needs `import scalaz._, Scalaz._`
  private def futuring[B, A](e: AerospikeException => B)(f: () => Option[A]): EitherT[Future, B, A] = {
    Future { f() } onComplete {
      case Success(res) => res match {
        case Some(x) => if (x.isInstanceOf[A]) x.right else e(new AerospikeException("not found")).left
        case None => e(new AerospikeException("null record returned")).left
      }
      case Failure(t) => t match {
        case t: AerospikeException => e(t).left
      }
    }
  }
```

File: src/main/scala/src/org/aerospiker/Settings.scala [hunk 2]
```diff
@@ -1,8 +1,12 @@
 package org.aerospiker
 
+case class Host(
+  name: String,
+  port: Int)
+
 // TODO: More support properties!
 case class Settings(
-  host: String = "127.0.0.1:3000",
+  host: Array[Host] = Array(Host("127.0.0.1", 3000)),
```
Comment (yanana) [inline src/main/scala/src/org/aerospiker/Settings.scala]: Why `Array`?

File: src/main/scala/src/org/aerospiker/Settings.scala [hunk 3]
```diff
@@ -1,8 +1,12 @@
 package org.aerospiker
 
+case class Host(
+  name: String,
+  port: Int)
+
 // TODO: More support properties!
```
Comment (yanana) [inline src/main/scala/src/org/aerospiker/Settings.scala]: "Support More properties!" instead is better.

File: src/main/scala/src/org/aerospiker/Operation.scala (hunk 1 above)
Comment (yanana) [inline src/main/scala/src/org/aerospiker/Operation.scala]: `futurize` may be better?

File: src/main/scala/src/org/aerospiker/Conversions.scala [hunk 4]
```diff
@@ -10,22 +10,33 @@ import java.lang.{ Integer => JInteger, Long => JLong, Double => JDouble, Float
...
+        case x: JLong => x: Long
         case x => x
       }
-      val bs = x.bins mapValues { convert(_) }
-      Record(
-        bins = bs.toMap,
-        generation = x.generation,
-        expiration = x.expiration)
+
+      x match {
```
Comment (yanana) [inline src/main/scala/src/org/aerospiker/Conversions.scala]:
``` scala
Option(x) map { a =>
  Option(a.bins) map { b =>
    Some(Record(
      bins = b.mapValues(convert(_)) toMap,
      generation = x.generation,
      expiration = x.expiration))
  }
}
```

File: src/main/scala/src/org/aerospiker/Settings.scala (hunk 2 above)
Comment (tkrs) [inline src/main/scala/src/org/aerospiker/Settings.scala]: No particular reason.

File: src/main/scala/src/org/aerospiker/Operation.scala (hunk 1 above)
Comment (tkrs) [inline src/main/scala/src/org/aerospiker/Operation.scala]: > futurize may be better?

That's true!

File: src/main/scala/src/org/aerospiker/Settings.scala (hunk 2 above)
Comment (yanana) [inline src/main/scala/src/org/aerospiker/Settings.scala]: `Seq` seems better to me.

File: src/main/scala/src/org/aerospiker/Operation.scala (hunk 1 above)
Comment (tkrs) [inline src/main/scala/src/org/aerospiker/Operation.scala]: Found the compile error in above code :disappointed_relieved:

File: src/main/scala/src/org/aerospiker/Conversions.scala (hunk 4 above)
Comment (tkrs) [inline src/main/scala/src/org/aerospiker/Conversions.scala]: Found the compile error in above code :disappointed_relieved:

File: src/main/scala/src/org/aerospiker/Settings.scala (hunk 3 above)
Comment (tkrs) [inline src/main/scala/src/org/aerospiker/Settings.scala]: That's true

File: src/main/scala/src/org/aerospiker/Conversions.scala (hunk 4 above)
Comment (yanana) [inline src/main/scala/src/org/aerospiker/Conversions.scala]: Sorry for wrong pointing out. I confirmed this works (but not tested):

``` scala
for {
    rec <- Option(x)
    b <- Option(rec.bins)
    r <- Some(Record(
      bins = b.mapValues(convert(_)) toMap,
      generation = rec.generation,
      expiration = rec.expiration ))
} yield r
```

File: src/main/scala/src/org/aerospiker/Operation.scala (hunk 1 above)
Comment (yanana) [inline src/main/scala/src/org/aerospiker/Operation.scala]:
``` scala
  private def futurize[B, A](e: AerospikeException => B)(f: () => Option[A]): EitherT[Future, B, A] =
    EitherT(Future {
      try {
        f() match {
          case Some(x) => x.right
          case None => e(new AerospikeException("null record responded")).left
        }
      } catch {
        case ex: AerospikeException => e(ex).left
      }
    })

```

**Comment by yanana** [comment]
LGTM

**Pull request merged**